gradio/networking.py

```diff
@@ -426,10 +426,20 @@ def queue_thread(path_to_local_server, test_mode=False):
             break
 
 
-def start_server(interface, server_name, server_port, auth=None, ssl=None):
-    port = get_first_available_port(
-        server_port, server_port + TRY_NUM_PORTS
-    )
+def start_server(interface, server_name, server_port=None, auth=None, ssl=None):
+    if server_port is None:  # if port is not specified, start at 7860 and search for first available port        
+        port = get_first_available_port(
+            INITIAL_PORT_VALUE, INITIAL_PORT_VALUE + TRY_NUM_PORTS
+        )
+    else:
+        try:
+            s = socket.socket()  # create a socket object
+            s.bind((LOCALHOST_NAME, server_port))  # Bind to the port to see if it's available (otherwise, raise OSError)
+            s.close()
+        except OSError:
+            raise OSError("Port {} is in use. If a gradio.Interface is running on the port, you can close() it or gradio.close_all().".format(server_port))
```

Comment:
`gradio.close_all()` doesn't seem to be freeing up the port. Can you try the workflow of assigning two interfaces the same port and then trying `close_all()` and using the port again? 

Would also make sense to write a test for this.